includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -549,9 +555,20 @@ private function get_lessons( $args ) {
 			$lessons_args['s'] = $args['search'];
 		}
 
-		// Using WP_Query as get_posts() doesn't support 'found_posts'
+		// Using WP_Query as get_posts() doesn't support 'found_posts'.
 		$lessons_query     = new WP_Query( apply_filters( 'sensei_analysis_overview_filter_lessons', $lessons_args ) );
 		$this->total_items = $lessons_query->found_posts;
+
+		if ( ! $include_completion_data ) {
+			return $lessons_query->posts;
+		}
+
+		$completions = Sensei_Lesson::get_lessons_users_completion_time_sum_with_count( array_column( $lessons_query->posts, 'ID' ) );
```

Comment:
Hm, interesting approach. Did you consider trying to alter the lessons query directly? That way you could organize the code a bit and save a few queries. You could use the `posts_clauses` filter for that. Here's a quick example that is not tested in-depth:

```php
public function add_days_to_complete_to_lessons_query( $clauses ) {
	global $wpdb;

	$clauses['fields']  .= ", sum( CEILING( timestampdiff( second, STR_TO_DATE( {$wpdb->commentmeta}.meta_value, '%Y-%m-%d %H:%i:%s' ), {$wpdb->comments}.comment_date ) / (24 * 60 * 60) )) as days_to_complete";
	$clauses['join']    .= " LEFT JOIN {$wpdb->comments} ON {$wpdb->comments}.comment_post_ID = {$wpdb->posts}.ID";
	$clauses['join']    .= " AND {$wpdb->comments}.comment_type IN ('sensei_lesson_status')";
	$clauses['join']    .= " AND {$wpdb->comments}.comment_approved IN ( 'complete', 'graded', 'passed', 'failed' )";
	$clauses['join']    .= " AND {$wpdb->comments}.comment_post_ID = {$wpdb->posts}.ID";
	$clauses['join']    .= " LEFT JOIN {$wpdb->commentmeta} ON {$wpdb->comments}.comment_ID = {$wpdb->commentmeta}.comment_id";
	$clauses['join']    .= " AND {$wpdb->commentmeta}.meta_key = 'start'";
	$clauses['groupby'] .= " {$wpdb->posts}.ID";

	return $clauses;
}

private function get_lessons( $args ) {
	...
	add_filter( 'posts_clauses', [ $this, 'add_days_to_complete_to_lessons_query' ] );
	$lessons_query = new WP_Query( $args );
	remove_filter( 'posts_clauses', [ $this, 'add_days_to_complete_to_lessons_query' ] );
	...
	return $lessons_query->posts; // Now we should have the calculated days_to_complete. 
}
```

I'm not sure if this will fit your case, so let me know. 🙂

Reply:
Thanks @m1r0 , that's a great approach, moving it to post_clauses saves a query and some calculations, also makes it a bit cleaner. Really liked it, updated it here https://github.com/Automattic/sensei/pull/4883/commits/79c47617ef6ec8e9b6ced62718165a3f11e80e84